src/lazy/binary/raw/v1_1/value.rs

```diff
@@ -150,8 +151,13 @@ impl<'top> LazyRawBinaryValue_1_1<'top> {
     /// or [`LazyStruct`](crate::lazy::struct::LazyStruct) that can be traversed to access the container's contents.
     pub fn read(&self) -> ValueParseResult<'top, BinaryEncoding_1_1> {
         if self.is_null() {
-            let raw_value_ref = RawValueRef::Null(self.ion_type());
-            return Ok(raw_value_ref);
+            let ion_type = if self.encoded_value.header.ion_type_code == OpcodeType::TypedNull {
+                let body = self.value_body()?;
+                ION_1_1_TYPED_NULL_TYPES[body[0] as usize]
+            } else {
+                self.ion_type()
```

Comment:
Since we know that the raw value is null but it's not using the `TypedNull` op code in this branch, then I think we could just return `IonType::Null` here. Maybe the compiler/optimizer is smart enough to figure that out, but I think that explicitly returning `IonType::Null` makes the meaning of the code a little more clear to the reader.